src/app/components/sites/manage-page/static.ts

```diff
@@ -46,9 +46,17 @@ export default class AppSitesManagePageStatic extends Vue {
 		try {
 			const response = await Api.sendRequest(
 				`/web/dash/sites/activate-primary-build/${this.site.id}`,
-				{}
+				{},
+				{ noErrorRedirect: true }
 			);
-			this.site.assign(response.site);
+
+			if (response.errors && response.errors.domain_in_use) {
+				Growls.error(this.$gettext('Domain is already in use in another site.'));
```

Comment:
I think we'd need to return after this, otherwise it will still continue.

Reply:
Ah yeah, it used to have some other logic at the end.